openlibrary/templates/type/edition/view.html

```diff
@@ -24,39 +24,49 @@
   $# We are on an editions page: an edition has been explicitly selected
   $ edition = page
 
+$ selected_provider = None
+$ selected_id = None
+$if query_param('edition'):
+  $if ':' in query_param('edition'):
+    $ [selected_provider, selected_id] = query_param('edition').split(':')
+  $else:
+    $ [selected_provider, selected_id] = ['ia', query_param('edition')]
+
 $ component_times['get_sorted_editions'] = time()
 
 $# Fetch a work's editions
 $# Used for loading the editions table
 $# Book availability of editions injected by bulk get_availability API
 $ edition_count = work.edition_count if work and work.edition_count else 1
-$ ebooks_only = query_param('mode') == 'ebooks' or (query_param('mode') != 'all' and edition_count > 10)
+$ ebooks_only = (query_param('mode') == 'ebooks') or (query_param('mode') != 'all' and edition_count > 10)
 
 $# For performance reasons, limit to 10 ebooks
 $# Tradeoff: limits our ability to select best edition
-$ editions_limit = None if not query_param('mode') else 10
+$ editions_limit = 10 if not query_param('mode') else None
+$# key ensures the current edition we're on or requested edition are fetched by get_sorted_editions
+$ keys = (edition and [edition.key]) or (selected_provider=="ia" and selected_id and [work.get_olid_by_ocaid(selected_id)])
```

Comment:
I think we will need something more generic now that the truncation is actually working.

```suggestion
$if selected_provider and selected_id:
    $ edition = get_provider_by_name(selected_provider).get_edition(selected_id)
```

where `get_edition` is basically the `get_many` call, but place it in the book_provider abstract class.